airflow/providers/amazon/aws/hooks/athena.py

```diff
@@ -260,3 +261,16 @@ def stop_query(self, query_execution_id: str) -> Dict:
         :return: dict
         """
         return self.get_conn().stop_query_execution(QueryExecutionId=query_execution_id)
+
+
+class AWSAthenaHook(AthenaHook):
+    """
+    This hook is deprecated.
+    Please use :class:`airflow.providers.amazon.aws.hooks.athena.AthenaHook`.
+    """
+
+    warnings.warn(
+        "This hook is deprecated. Please use `airflow.providers.amazon.aws.hooks.athena.AthenaHook`.",
+        DeprecationWarning,
+        stacklevel=2,
+    )
```

Comment:
```suggestion
class AWSAthenaHook(AthenaHook):
    """
    This hook is deprecated.
    Please use :class:`airflow.providers.amazon.aws.hooks.athena.AthenaHook`.
    """

    def __init__(self, *args, **kwargs):
        warnings.warn(
            "This hook is deprecated. Please use `airflow.providers.amazon.aws.hooks.athena.AthenaHook`.",
            DeprecationWarning,
            stacklevel=2,
        )
        super().__init__(*args, **kwargs)
```

Having the warning directly in the class body would cause the message to be emitted _on import time_, which is not correct.

Reply:
Sorry, just saw these comments after I made those changes.  I'll move those into inits.